opensearch-project/OpenSearch-Dashboards
# [Workspace]Fix flights sample data copy in workspace assets page

### Description


This PR fixes an issue where the flights sample data failed to copy to other workspaces due to missing index patterns in the workspace assets page. The root cause of this issue was that the index under the import sample data saved object's search source filter metadata was not regenerated with the data source ID and workspace ID. The [visualization migration function](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/visualizations/server/saved_objects/visualization_migrations.ts#L48-L70) adds used index patterns from the imported sample data saved object's search source. After that, when we call the duplicate saved objects API to copy assets, it returns a `missing_references` error, and the assets are not copied. 
To address this issue, we need to add index pattern ID regeneration logic in the [import sample data util](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/home/server/services/sample_data/data_sets/util.ts#L44).

## Screenshot


No UI changes

## Testing the changes



- Clone branch code and run `yarn osd bootstrap --single-version loose`
- Add below configs in `config/opensearch_dashboards.yml`
```
savedObjects.permission.enabled: true
workspace.enabled: true
uiSettings:
  overrides:
    'home:useNewHomePage': true
```
- Run `yarn start --no-base-path`
- Login with admin user
- Create a observability workspace named test1
- Create a observability workspace named test2
- Visit sample data page inside workspace test2
- Click "Add data" inside flights sample data card
- Visit workspace assets page
- Click the checkbox on the left of "[Flights] Delay Buckets"
- Click "Copy to ..." and select "test1" as target workspace
- Click "Copy" button, the selected assets should be copied
- Visit assets page inside test1 workspace
- The "[Flights] Delay Buckets" should be exists

## Changelog

- fix: [Workspace]Fix flights sample data copy in workspace assets page

### Check List
